web/src/components/common/DeadlineTag.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import {formatDistanceToNowStrict, isPast} from 'date-fns';
+import styled from 'styled-components';
+
+const StyledDeadline = styled.span`
+  display: flex;
+  justify-content: flex-end;
+
+  color: var(--dark-gray);
+  font-size: 0.9em;
+`;
+
+const RedText = styled.span`
+  color: var(--red);
+`;
+
+interface DeadlineTagProps {
+  deadline: string;
+}
+
+/**
+ * Component that displays the deadline of a listing.
+ * Displays a red text with number of days left if deadline
+ * is not over.
+ * Displays number of days since deadline if listing has ended.
+ */
+const DeadlineTag: React.FC<DeadlineTagProps> = ({deadline}) => {
+  const deadlineDate = new Date(deadline);
+  const tag = formatDistanceToNowStrict(deadlineDate, {unit: 'day'});
+  return (
+    <StyledDeadline>
+      {isPast(deadlineDate) && <>Ended {tag} ago</>}
```

Comment:
As we do not have tests at all, in the description could we have a screenshot that displays one listing card that has already passing the deadline and another one that is not passing the deadline to see the different looks of both sides? Currently both cards shown are not passing deadline.

Reply:
@xlai20  Hmm ok I can add an ended listing to the explore page! But now that I think about it, I don't think we should be even showing past listings on the Explore page at all.

Reply:
@xlai20 I can extract this string out to a utils function and add tests too!